server/controllers/medical/patients/index.js

```diff
@@ -668,23 +671,25 @@ function latestInvoice (req, res, next) {
     return q.all([execSql, execSql2, execSql3]);
   })
   .spread(function (invoices, payments, invoicesLength) {
-    var numberPayment = payments[0].numberPayment;
-    invoice = {
-      uuid            : invoiceLatest.uuid,
-      debtor_uuid     : invoiceLatest.debtor_uuid,
-      numberPayment   : numberPayment,
-      date            : invoiceLatest.date,
-      cost            : invoiceLatest.cost,
-      display_name    : invoiceLatest.display_name,
-      uid             : invoices[0].uid,
-      reference       : invoices[0].reference,
-      credit          : invoices[0].credit,
-      debit           : invoices[0].debit,
-      balance         : invoices[0].balance,
-      entity_uuid     : invoices[0].entity_uuid,
-      invoicesLength  : invoicesLength[0].invoicesLength
-    };
-
+    var numberPayment = payments ? payments[0].numberPayment : 0; //what is his meaning?
+    if(invoices && invoicesLength){
```

Comment:
Hmm.  Instead of trying to manipulate this current promise chain, it would be cleaner to separate the logic into a helper function.  Like this:
```js
function latestInvoice(req, res, next) {
  // declare variables, etc..

  return db.exec(sql, [uid])
    .then(rows => {
      let hasLatestInvoice = rows.length > 0;
      let latestInvoice = rows[0];

      // fake promise to simply skip the latest invoice loading.
      let skip = q({});  

      return hasLatestInvoice ?
        loadLatestInvoice(latestInvoice) : 
        skip;
    })
    .then(res => {
      res.status(200).json(invoice);
    })
    .catch(next)
    .done();
});

function loadLatestInvoice(invoice) {
   // in this function, do the db.exec().spread(...);
}
```

Does that make sense?  Then you will not be forced to write `db.exec(sql, [null, null, null])` and make the database/server do unnecessary work.